lib/tests/regex-with-escapes.test.ts

```diff
@@ -8,11 +8,11 @@ test("regex-with-escapes", () => {
             properties: {
                 str: { 
                     type: "string",
-                    pattern: "^\/$"
+                    pattern: "^/$"
                 },
             }
         }})
     ).toMatchInlineSnapshot(
-        '"z.object({ str: z.string().regex(/^\/$/) }).partial().passthrough()"'
+        '"z.object({ str: z.string().regex(/^\\/$/) }).partial().passthrough()"'
```

Comment:
The point of this test is that we want the resulting code to include the escape character `\` before the `/`, otherwise the `/` is interpreted as the end of the regex!

Since the `\` wasn't escaped in the string definition, same as above it wasn't being included in the string in memory.

```
console.log("/^\/$/")
/^/$/
console.log("/^\\/$/")
/^\/$/
```